vue/components/organisms/personalization/personalization.vue

```diff
@@ -169,6 +173,16 @@ export const Personalization = {
         },
         hasPersonalization() {
             return this.$store.state.hasPersonalization;
+        },
+        allowApply() {

```

Comment:
The approach is in the right place, but let's go with a different strategy: using `invalidState`. This must return `true` for all cases where the SDK would throw an error:
- https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/base/ripe.js#L711
- https://github.com/ripe-tech/ripe-sdk/blob/master/src/js/base/ripe.js#L790

This means it must verify both initials+engraving as well as initialsExtra

Reply:
>The approach is in the right place, but let's go with a different strategy: using invalidState. This must return true for all cases where the SDK would throw an error:
I tested and unfortunately it appears those errors are never thrown, even if we apply a personalization without an engraving. Could I be doing something wrong?
>This means it must verify both initials+engraving as well as initialsExtra
As it is are verying legacy Initials+engraving as well as initialsExtra from the `state`

Reply:
The problem is that right now having no initials disables the button, which is not desirable. The button must only be disabled when you have invalid initials, using the SDK definition.

I'm not saying to use the SDK, just to replicate the logic that is there. You can even place that function inside the logic mixin.

Reply:
> The problem is that right now having no initials disables the button, which is not desirable. The button must only be disabled when you have invalid initials, using the SDK definition.

Having no initials disables the apply button because a personalization is never valid if it has no intials or am I wrong?
Even the SDK is checking for that case:
```
        if (!value.initials && value.engraving) {
            throw new Error(`Engraving set without initials for group ${key}`);
        }
``` 

> I'm not saying to use the SDK, just to replicate the logic that is there. You can even place that function inside the logic mixin.
Understood.

Reply:
What that line means is that it is invalid if it has no initials but it has engraving. It's a boolean implication: `initials => engraving`, meaning `if it has initials, it must have engraving`.